### PR TITLE
Move dolt system tables to dolt schema for doltgres

### PR DESCRIPTION
**Pull request by tbantle22**

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>e20f216</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>e20f216</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>725d752</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>725d752</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>64e2692</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>64e2692</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>0809357</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>0809357</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>674d29f</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>674d29f</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/database.go [hunk 1]
```diff
@@ -466,23 +494,29 @@ func (db Database) getTableInsensitive(ctx *sql.Context, head *doltdb.Commit, ds
...
 
-		dt, found = dtables.NewStatusTable(ctx, db.ddb, ws, adapter), true
+			dt, found = dtables.NewStatusTable(ctx, db.ddb, ws, adapter), true
+		}
 	case doltdb.MergeStatusTableName:
 		dt, found = dtables.NewMergeStatusTable(db.RevisionQualifiedName()), true
-	case doltdb.TagsTableName:
-		dt, found = dtables.NewTagsTable(ctx, db.ddb), true
+	case doltdb.GetTagsTableName(), doltdb.TagsTableName:
+		// TODO: This should be moved once all the system tables are moved to the dolt schema
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:516]: This comment here and elsewhere seems out of date

File: go/libraries/doltcore/sqle/database.go [hunk 2]
```diff
@@ -314,6 +314,28 @@ func (db Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, a
...
+// isDoltSystemTable checks if the current table is a dolt system table. If
+// using the search path, also checks that dolt is the current schema or exists
+// on the search path.
+func (db Database) isDoltSystemTable(ctx *sql.Context, tableName string) bool {
+	if doltdb.HasDoltPrefix(tableName) || !resolve.UseSearchPath || db.schemaName == "dolt" {
+		return true
+	}
+
+	if db.schemaName == "" {
+		schemasToSearch, err := resolve.SearchPath(ctx)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:326]: This block is subtly wrong. If the user has a table named e.g. `log` and that schema is on the search path before the `dolt` schema, that name must resolve to their table. Should add a test for this case.

Comment (zachmu) [reply go/libraries/doltcore/sqle/database.go:326]: You should probably write `resolve.SystemTable(ctx, tableName)` that handles this logic and call it here.

File: go/libraries/doltcore/sqle/database.go [hunk 3]
```diff
@@ -314,6 +314,28 @@ func (db Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, a
 	}
 }
 
+// isDoltSystemTable checks if the current table is a dolt system table. If
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:317]: Comment is subtly wrong, see comments in code below

**Review by zachmu — APPROVED** [review]
LGTM with a couple caveats.

There's also some logic and a comment in `Database.resolveUserTable` that should now be cleaned up.

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ad541d7</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ad541d7</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/resolve/search_path.go [hunk 4]
```diff
@@ -78,3 +78,36 @@ func FirstExistingSchemaOnSearchPath(ctx *sql.Context, root doltdb.RootValue) (s
...
+			return false, nil
+		}
+		for _, schemaName := range schemasToSearch {
+			if schemaName != "dolt" {
+				tablesInSchema, err := root.GetTableNames(ctx, schemaName)
+				if err != nil {
+					return false, err
+				}
+				for _, table := range tablesInSchema {
+					if table == tableName.Name {
```
[the remark this answers is not in the document]
Comment (zachmu) [reply go/libraries/doltcore/sqle/resolve/search_path.go:100]: Need a test for this

**Review by zachmu — APPROVED** [review]
LGTM, couple notes

File: go/libraries/doltcore/sqle/database.go [hunk 5]
```diff
@@ -314,6 +314,10 @@ func (db Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, a
 	}
 }
 
+func isDoltSystemTable(tableName string) bool {
+	return doltdb.HasDoltPrefix(tableName) || !resolve.UseSearchPath
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:318]: This seems off, won't these always both be true in the context?

Seems like maybe call to this method should just be replaced with `!resolve.UseSearchPath`, would be more explicit what was going on

File: go/libraries/doltcore/sqle/database.go [hunk 6]
```diff
@@ -419,17 +423,24 @@ func (db Database) getTableInsensitive(ctx *sql.Context, head *doltdb.Commit, ds
...
 	switch lwrName {
-	case doltdb.LogTableName:
-		if head == nil {
-			var err error
-			head, err = ds.GetHeadCommit(ctx, db.RevisionQualifiedName())
-			if err != nil {
-				return nil, false, err
-			}
+	case doltdb.GetLogTableName(), doltdb.LogTableName:
+		systemTable, err := resolve.IsSystemTable(ctx, tname, root)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:429]: Probably would call this isDoltgresSystemTable to make this super explicit

**Review by zachmu — APPROVED** [review]
Couple final small comments

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ac54d05</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ac54d05</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**